cranelift/codegen/src/isa/s390x/inst.isle

```diff
@@ -3478,13 +3475,14 @@
         dst))
 
 ;; Lower a boolean condition to a boolean type.  The value used to represent
-;; "true" is -1 for all result types except for $B1, which uses 1.
+;; "true" is -1 for all result types except for $I8, which uses 1.
 (decl lower_bool (Type ProducesBool) Reg)
-(rule (lower_bool $B1 cond) (select_bool_imm $B1 cond 1 0))
-(rule (lower_bool $B8 cond) (select_bool_imm $B8 cond -1 0))
-(rule (lower_bool $B16 cond) (select_bool_imm $B16 cond -1 0))
-(rule (lower_bool $B32 cond) (select_bool_imm $B32 cond -1 0))
-(rule (lower_bool $B64 cond) (select_bool_imm $B64 cond -1 0))
+(rule (lower_bool $I8 cond) (select_bool_imm $I8 cond 1 0))
+
+;; TODO: do we need these cases anymore if B8..B128 are missing?
```

Comment:
Hmm.  This helper is only called for possible scalar result types of `icmp` and `fcmp`.   These are specified as `&Int.as_bool()` and `&Float.as_bool()` so I thought I needed to handle multiple widths here.  But looking at this again, actually the only scalar return type of `as_bool` has always been `$B1` (and is now `$I8`), so it seems this is only type we need to handle here in the first place.